File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(!root) return {};
        queue<TreeNode*>q;
        vector<vector<int>>v;
        q.push(root);
        while(!q.empty()){
            vector<int>temp;
            int size = q.size();
            for(int i = 0 ; i < size ; ++i){
                TreeNode* head = q.front();
                q.pop();
                temp.push_back(head->val);
                if(head->left) q.push(head->left);
                if(head->right) q.push(head->right);
            }
            v.push_back(temp);
        }
        return v;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root==null) return new ArrayList<List<Integer>>();
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        Queue<TreeNode>q = new LinkedList<>();
        q.offer(root);
        while(!q.isEmpty()){
            List<Integer>list = new ArrayList<Integer>();
            int size = q.size();
            for(int i = 0 ; i < size ; ++i){
                 TreeNode head = q.poll();
                 list.add(head.val);
                if(head.left!=null){ 
                    q.offer(head.left);
                }
                if(head.right!=null){
                    q.offer(head.right);
                } 
            }
            res.add(list);
        }
        return res;
    }
}